src/clusterfuzz/_internal/build_management/build_archive.py

```diff
@@ -328,3 +331,34 @@ def open(archive_path: str) -> BuildArchive:
   if reader.file_exists(args_gn_path):
     return ChromeBuildArchive(reader)
   return DefaultBuildArchive(reader)
+
+
+def open_uri(uri: str) -> BuildArchive:
+  """Opens a build archive over HTTP. This is only compatible with chromium as
+  of now.
+
+  Args:
+      uri: the gs:// storage URI.
+
+  Returns:
+      the build archive.
+  """
+  assert unzip_over_http_compatible(uri)
+  match = CHROMIUM_GS_MATCHER.match(uri)
+  uri = f'https://commondatastorage.googleapis.com/{match.group(1)}'
+  reader = archive.ZipArchiveReader(archive.HttpZipFile(uri))
+  args_gn_path = os.path.join(reader.root_dir(), 'args.gn')
+  if reader.file_exists(args_gn_path):
+    return ChromeBuildArchive(reader)
+  return DefaultBuildArchive(reader)
+
+
+def unzip_over_http_compatible(build_url: str) -> bool:
+  """Whether the build URL is compatible with unzipping over HTTP.
+  As for now, we're only checking for chromium compatible URLs.
+  """
+  match = CHROMIUM_GS_MATCHER.match(build_url)
+  if not match:
+    return False
+  uri = f'https://commondatastorage.googleapis.com/{match.group(1)}'
```

Comment:
We're matching against chromium buckets and extracting the URI twice: once in the assertion that calls this function and again immediately after. If we drop the chromium requirement, then we can simply move the `is_uri_compatible()` assertion into `HttpZipFile`'s constructor.